arangod/Aql/ExecutionBlock.cpp

```diff
@@ -397,15 +393,15 @@ int ExecutionBlock::getOrSkipSome(size_t atLeast, size_t atMost, bool skipping,
   // if _buffer.size() is > 0 then _pos points to a valid place . . .
   BlockCollector collector(&_engine->_itemBlockManager);
 
-  while (skipped < atLeast) {
+  while (skipped < atMost) {
     if (_buffer.empty()) {
       if (skipping) {
         size_t numActuallySkipped = 0;
-        _dependencies[0]->skip(atLeast - skipped, numActuallySkipped);
-        skipped = atLeast;
+        _dependencies[0]->skip(atMost - skipped, numActuallySkipped);
+        skipped = atMost;
```

Comment:
i think this assignment is not correct. (it was not in original either...)
Shouldn't it be:
skipped += numActuallySkipped